api/apps/geoprocessing/src/modules/scenarios/cost-template/cost-template-generator.ts

```diff
@@ -0,0 +1,121 @@
+import { Injectable } from '@nestjs/common';
+import { EntityManager } from 'typeorm';
+import { InjectEntityManager } from '@nestjs/typeorm';
+import * as fs from 'fs';
+import * as path from 'path';
+import * as uuid from 'uuid';
+import * as archiver from 'archiver';
+import { runCommandsXL } from 'mapshaper';
+import {
+  ArtifactType,
+  ScenarioCostSurfaceRepository,
+} from '@marxan/scenario-cost-surface';
+import { geoprocessingConnections } from '@marxan-geoprocessing/ormconfig';
+import { AppConfig } from '@marxan-geoprocessing/utils/config.utils';
+import { FeatureCollectionWrapper } from './feature-collection-wrapper';
+import { assertDefined } from '@marxan/utils';
+
+@Injectable()
+export class CostTemplateGenerator {
+  constructor(
+    @InjectEntityManager(geoprocessingConnections.default)
+    private readonly entityManager: EntityManager,
+    private readonly repository: ScenarioCostSurfaceRepository,
+  ) {}
+
+  async createTemplateShapefile(scenarioId: string) {
+    const storagePath = AppConfig.get<string>(
+      'storage.sharedFileStorage.localPath',
+    );
+    assertDefined(storagePath);
```

Comment:
It already throws if not found (and no default value is provided)

Reply:
the typing is an issue